app/controllers/movies_controller.rb

```diff
@@ -66,4 +71,23 @@ def load_episode
     flash[:danger] = t "not_found.episode"
     redirect_to page_404_path
   end
+
+  def view_counter movie
+    user_movie = UserMovie.find_by user_id: current_user.id,
+      movie_id: movie.id
+    if user_movie
+      UserMovie.transaction do
+        user_movie.views = user_movie.views + Settings.step.one
+        user_movie.save
+      end
+    else
+      UserMovie.transaction do
+        user_movie = UserMovie.create(user_id: current_user.id,
+          movie_id: movie.id, views: Settings.default_views)
+      end
+    end
+    Movie.transaction do
+      movie.update_attribute :views, user_movie.views
+    end
```

Comment:
Ahihi, 1 traction phủ từ line 78 -> 91 nhé, chứ như ri thì cũng như không

usermovie vẫn dc update dù move có bị fail ...:p

Reply:
dùng `update_attributes!` thay vì `update_attribute` em nhé

Vì mặc dù là update_attribute có raise exception, nhưng chỉ khi field em update chỉ có quyền đọc chứ ko có quyền edit :p 
https://stackoverflow.com/questions/21028625/rails-is-there-an-update-or-update-attributes-that-throws-exception-on-failure